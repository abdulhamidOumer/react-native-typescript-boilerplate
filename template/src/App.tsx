/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the React Native TypeScript Boilerplate template
 * https://github.com/abdulhamidOumer/react-native-typescript-boilerplate
 *
 * @format
 */

import React, { useEffect } from 'react';
import { SafeAreaView, StatusBar, useColorScheme } from 'react-native';

import { Provider } from 'react-redux';
import SplashScreen from 'react-native-splash-screen';
import { configureAppStore } from 'store/configureStore';

import { Colors } from 'react-native/Libraries/NewAppScreen';
import MainNavigator from 'app/Navigations';

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
    flex: 1,
  };

  const store = configureAppStore();

  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <Provider store={store}>
      <SafeAreaView style={backgroundStyle}>
        <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
        <MainNavigator />
      </SafeAreaView>
    </Provider>
  );
};

export default App;
