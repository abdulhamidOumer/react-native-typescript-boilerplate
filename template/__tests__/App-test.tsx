/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../src/App';

// Note: test renderer must be required after react-native.
import { createRenderer } from 'react-test-renderer/shallow';

const renderer = createRenderer();

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    // renderer.render(<App />);
  });
});
