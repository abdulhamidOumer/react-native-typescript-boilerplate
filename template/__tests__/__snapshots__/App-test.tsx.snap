// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "createReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
    }
  }
>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <StatusBar
      animated={false}
      barStyle="dark-content"
      showHideTransition="fade"
    />
    <ScrollView
      contentInsetAdjustmentBehavior="automatic"
      style={
        Object {
          "backgroundColor": "#F3F3F3",
        }
      }
    >
      <Header />
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
          }
        }
      >
        <Section
          title="Step One"
        >
          Edit 
          <Text
            style={
              Object {
                "fontWeight": "700",
              }
            }
          >
            App.tsx
          </Text>
           to change this screen and then come back to see your edits.
        </Section>
        <Section
          title="See Your Changes"
        >
          <ios />
        </Section>
        <Section
          title="Debug"
        >
          <ios />
        </Section>
        <Section
          title="Learn More"
        >
          Read the docs to discover what to do next:
        </Section>
        <LinkList />
      </View>
    </ScrollView>
  </ForwardRef(SafeAreaView)>
</Provider>
`;
